### ISSUE TITLE
Hardlink source gone after clearing cache

### ISSUE THREAD

**Issue by solverat** [pimcore/pimcore#723]
1.) create a hardlink element on root (eg. /test)
2.) link it with a document and check both checkboxes
3.) save and open link (eg. www.site.com/test)
4.) clear cache ("_Clear Cache_")
5.) reopen link ( eg. www.site.com/test )
6.) frontend will stop working, throws fatal error
```
Call to undefined method Pimcore\Model\Document\Hardlink\Dao::getElements()
```
only resaving the document will remove this error.

**Comment by dpfaffenbauer** [comment]
can confirm it.

1. doesn't need to be on root

only re-opening and re-saving solved the problem.

**Comment by brusch** [comment]
It works again after a reload, ... very strange.

**Comment by brusch** [comment]
Caused by fix for #655
